src/gui/widgets/LcdWidget.cpp

```diff
@@ -177,21 +185,24 @@ void LcdWidget::paintEvent( QPaintEvent* )
 	}
 
 	// Right Margin
-	p.drawPixmap( QRect( 0, 0, m_marginWidth-1, m_cellHeight ), *m_lcdPixmap, 
-			QRect( charsPerPixmap*m_cellWidth, isEnabled()?0:m_cellHeight,
-				m_cellWidth / 2, m_cellHeight ) );
+	p.drawPixmap(QRect(0, 0, m_seamlessRight ? m_marginWidth : m_marginWidth - 1, m_cellHeight),
+		*m_lcdPixmap,
+		QRect(charsPerPixmap * m_cellWidth, isEnabled() ? 0 : m_cellHeight, m_cellWidth / 2, m_cellHeight));
 
 
 	p.restore();
 
 	// Border
-	QStyleOptionFrame opt;
-	opt.initFrom( this );
-	opt.state = QStyle::State_Sunken;
-	opt.rect = QRect( 0, 0, m_cellWidth * m_numDigits + (margin+m_marginWidth)*2 - 1,
-			m_cellHeight + (margin*2) );
+	if (!m_seamlessLeft && !m_seamlessRight)
+	{
+		QStyleOptionFrame opt;
+		opt.initFrom(this);
+		opt.state = QStyle::State_Sunken;
+		opt.rect = QRect(0, 0, m_cellWidth * m_numDigits + (margin + m_marginWidth) * 2 - 1,
+			m_cellHeight + (margin * 2));
```

Comment:
I'm not sure if I read it correctly, but wouldn't that mean that all "vanilla" `LcdWidget`s would have no borders?

Reply:
No, I just meant that on the regular LCD Widget, the borders are being drawn on the `LcdWidget` class (`LcdWidget::paintEvent`, where this comment was made) but on the Float LCD Widget they are being drawn on the `LcdFloatSpinBox` class (`LcdFloatSpinBox::paintEvent`). As far as I understood, LcdWidget is the LCD number while the LcdSpinBox is the widget itself, that contains the LCD number. LcdFloatSpinBox for example has 2 LcdWidgets.

I was just pointing out it feels inconsistent to draw borders on the LcdWidgets on the regular spin box and on the LcdFloatSpinBox for the float spin box. We probably should do the border drawing on the spin box classes (both `LcdSpinBox` and `LcdFloatSpinBox`).

Reply:
> We probably should do the border drawing on the spin box classes (both LcdSpinBox and LcdFloatSpinBox).

But that would mean `LcdWidget` is left with no border-drawing code, hence my worry that `LcdWidgets` will not have borders if we do that. They are currently used in the FX mixer (where a nuber is needed, but you cannot change it), so there would be no borders around the green numbers on the FX mixer. Either that, or I'm getting really confused. :D

`LcdFloatSpinBox` does border drawing simply because it needs to visually join two widgets that were not originally designed to be joined. That's also why there are the new "seamless" parameters in `LcdWidget`. It all feels a bit like a hack, all the three LCD widgets could be probably rewritten in some way to make it cleaner, but I felt that is already way out of scope of the original Microtuner (I generally try to avoid touching existing code when not necessary, not to mention big rewrites like this).

Reply:
Oh, I didn't realize that `LcdWidget` was used by other widgets besides the Spin Box. Well, in that case it's probably a big rewrite to move the border to the outer widgets: Even though it makes sense it's out of the scope of this PR. I'm not sure it's worth it adding a `TODO` comment for the future, but you can ignore this comment then!

Reply:
Could a comment be added at the beginning about the border drawing, just for future reference? Something like:
```
// When either the right or left is seamless we are in a LcdFloatSpinBox
// and the border drawing will be done by the float spin box class instead
// TODO: Possibly move the frame drawing to the outer widget classes
// (i.e.: Spin boxes) for consistency
```

I know this sounds unimportant, but if in the future someone decides to change the frame drawing and isn't aware of that, the frames of regular spin boxes and float spin boxes won't match. The `TODO` also serves as a reference for possibly coming up with an alternative in the future.

Reply:
I added a comment explaining why is the border drawing skipped. I left out the "TODO", because I still do not understand where do you propose to "move" the code. If it is moved, it is no longer there, which breaks `LcdWidget` borders in the mixer. So we can only _copy_ it to `LcdSpinBox`, which does not make sense, it only adds more duplicate code that needs to be maintained.

Reply:
I don't know, maybe someone else can step in to give an opinion too, I could be wrong after all. For me it feels like bad design that a widget contains another one (SpinBox contains LcdWidgets) but in some cases the inner widget (LcdWidget) draws the frame and in other cases the outer widget (Float Spinbox) draws the frame. In a ideal scenario either the inner OR the outer widget should be responsible by the frame. That's all I was suggesting, that in the future we move the drawing of the frame to the outer widgets (LcdFloatSpinBox, LcdSpinBox, FxLineLcdSpinBox, etc) or that we come up with a design where we don't have it being drawn in different places.

I understand this is out of the scope of this PR (and obviously not its fault by any means), I just thought about adding a `TODO` comment so we could remember about maybe fixing it in the future. Just to let you know, this is by no means holding my approval (as I mentioned in Discord, I'm mostly waiting on @DomClark regarding the `update` method name, the aside the PR LGTM).